src/vs/workbench/parts/localizations/electron-browser/localizations.contribution.ts

```diff
@@ -141,19 +141,15 @@ export class LocalizationWorkbenchContribution extends Disposable implements IWo
 					return;
 				}
 
-				const extensionIdPostfix = this.getPossibleChineseMapping(locale);
-				const ceintlExtensionSearch = this.galleryService.query({ names: [`MS-CEINTL.vscode-language-pack-${extensionIdPostfix}`], pageSize: 1 });
-				const tagSearch = this.galleryService.query({ text: `tag:lp-${locale}`, pageSize: 1 });
-
-				TPromise.join([ceintlExtensionSearch, tagSearch]).then(([ceintlResult, tagResult]) => {
-					if (ceintlResult.total === 0 && tagResult.total === 0) {
+				this.galleryService.query({ text: `tag:lp-${locale}` }).then(tagResult => {
+					if (tagResult.total === 0) {
 						return;
 					}
 
-					const extensionToInstall = ceintlResult.total === 1 ? ceintlResult.firstPage[0] : tagResult.total === 1 ? tagResult.firstPage[0] : null;
-					const extensionToFetchTranslationsFrom = extensionToInstall || (tagResult.total > 0 ? tagResult.firstPage[0] : null);
+					const extensionToInstall = tagResult.total === 1 ? tagResult.firstPage[0] : tagResult.firstPage.filter(e => e.publisher === 'MS-CEINTL' && e.name.indexOf('vscode-language-pack') === 0)[0];
+					const extensionToFetchTranslationsFrom = extensionToInstall || tagResult.firstPage[0];
```

Comment:
One minor suggestion, how about merging above two lines as follows

`const extensionToFetchTranslationsFrom = tagResult.firstPage.filter(e => e.publisher === 'MS-CEINTL' && e.name.indexOf('vscode-language-pack') === 0)[0] || tagResult.firstPage[0];`

It's just cosmetic and i will leave it to you.

Reply:
`extensionToInstall` and `extensionToFetchTranslationsFrom` are intentionally kept as 2 separate variables. When we find the language pack from Microsoft or of there is only 1 language pack out there for the given locale, then both the variables will have the same value.

But when there are multiple language packs for the given locale and there is none from Microsoft, then we need `extensionToInstall` to be null so that we later ask user to search instead of install